### PR TITLE
PostgreSQL support

### PR DESCRIPTION
**Pull request by Pedrock**
I have implemented PostgreSQL support (version >= 9.5). 

## Description
I had to implement upsert in peewee using the new postgresql command ON CONFLICT. This custom peewee is in the lib folder, so it was removed from the requirements.txt. I am only inserting the rows one by one because the 120 number was giving me some deadlocks in postgresql. I think this is related to the ON CONFLICT command.

## Motivation and Context
This is just an alternative to MySQL.

## How Has This Been Tested?
I have been testing with a large number of threads and a large area with success.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pogom/models.py [hunk 1]
```diff
@@ -35,6 +35,13 @@ def init_database():
...
             host=args.db_host)
         log.info('Connecting to MySQL database on {}.'.format(args.db_host))
+    elif args.db_type == 'postgresql':
+        db = PostgresqlDatabase(
+            args.db_name, 
+            user=args.db_user, 
+            password=args.db_pass, 
+            host=args.db_host)
+        log.info('Connecting to PostgreSQL database on {}.'.format(args.db_host))
     else:
```
Comment (Ashex1) [inline pogom/models.py]: Can you add a test for the sqlite file here?

```python
elif os.path.isfile(args.db):
  db = SqliteDatabase(args.db)
   log.info('Connecting to local SQLLite database.')
else:
  raise ValueError('No valid Database configuration provided')
```

**Comment by invisiblek** [comment]
what's with the addition of peewee.py?

**Comment by Pedrock** [comment]
@invisiblek It is explained above. I had to implement upsert for postgresql.

**Comment by marauder37** [comment]
So this requires pgsql 9.5? Upsert, what a pain

**Comment by Pedrock** [comment]
@marauder37 Yes, this requires pgsql 9.5 or above. ON CONFLICT was added in 9.5.

**Comment by arthurkay** [comment]
+1 for external db support, shared sqlite sucks

**Comment by invisiblek** [comment]
@arthurkay MySQL is already supported...

**Comment by Pedrock** [comment]
I just split my commit into 2 in order to make my changes to the peewee library visible.

**Comment by coleifer** [comment]
I'll take a look at merging your peewee changes (or their functional equivalent) in the next few days.

**Comment by Pedrock** [comment]
@coleifer Nice! :)

**Pull request closed without merging**